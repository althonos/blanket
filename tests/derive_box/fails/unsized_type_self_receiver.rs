use blanket::blanket;
use impls::impls;

#[blanket(derive(Box))]
pub trait StringLike {
    fn into_utf8(self) -> Vec<u8>;
}

impl StringLike for str {
    fn into_utf8(self) -> Vec<u8> {
        self.as_bytes().into()
    }
}

fn main() {
    assert!(impls!(str:      StringLike));
    assert!(impls!(Box<str>: StringLike));
}
