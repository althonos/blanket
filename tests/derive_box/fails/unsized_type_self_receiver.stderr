error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> tests/derive_box/fails/unsized_type_self_receiver.rs:10:18
   |
10 |     fn into_utf8(self) -> Vec<u8> {
   |                  ^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `str`
   = help: unsized fn params are gated as an unstable feature
help: function arguments must have a statically known size, borrowed types always have a known size
   |
10 |     fn into_utf8(&self) -> Vec<u8> {
   |                  +
